clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/schema/HDFSRmdSchemaSource.java

```diff
@@ -0,0 +1,127 @@
+package com.linkedin.venice.hadoop.schema;
+
+import com.linkedin.venice.annotation.NotThreadsafe;
+import com.linkedin.venice.controllerapi.ControllerClient;
+import com.linkedin.venice.controllerapi.MultiSchemaResponse;
+import com.linkedin.venice.schema.rmd.RmdVersionId;
+import com.linkedin.venice.utils.Utils;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.avro.Schema;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * HDFSRmdSchemaSource caches the RMD schemes for a given store on the HDFS and serve them during the Mapper stage
+ * It will make API calls on Venice controller to retrieve all RMD schemes.
+ * <p>
+ * Borrowed some codes from HDFSSchemaSource. The difference is HDFSRmdSchemaSource is created for a single store.
+ */
+@NotThreadsafe
+public class HDFSRmdSchemaSource implements RmdSchemaSource, AutoCloseable {
+  private static final Logger LOGGER = LogManager.getLogger(HDFSRmdSchemaSource.class);
+  private static final String UNDERSCORE = "_";
+  private static final String SEPARATOR = "/";
+
+  private final String storeName;
+  private final FileSystem fs;
+  private final Path schemaDir;
+
+  public HDFSRmdSchemaSource(final String schemaDir, final String storeName) throws IOException {
+    Configuration conf = new Configuration();
+    this.fs = FileSystem.get(conf);
+    this.schemaDir = new Path(schemaDir);
+    if (!fs.exists(this.schemaDir)) {
+      fs.mkdirs(this.schemaDir);
+    }
+    this.storeName = storeName;
+  }
+
+  public HDFSRmdSchemaSource(final String schemaDirSuffix) throws IOException {
```

Comment:
Do we need this constructor?
I assume `storeName` is always known in the map-reduce job.

Reply:
Removed.